File: DataStruct/python/stack.py
import sys


class Node:

    def __init__(self, data=None):
        self.data = data  # assign data
        self.next = None  # initialize next as null


class Stack:

    def __init__(self):
        self.top = None

    def getTop(self):
        if not self.top:
            print('Top is Null')
            return
        print(self.top.data)

    def isStackEmpty(self):
        if not self.top:
            return True
        return False

    def push(self, item, verbose=False):
        new_Node = Node(item)
        if self.top is None:
            self.top = new_Node
        else:
            new_Node.next = self.top
            self.top = new_Node
        if verbose:
            self.printStack()
        return

    def pop(self, verbose=False):
        if not self.isStackEmpty():
            temp = Node()
            temp = self.top
            self.top = self.top.next
            del temp
            if verbose:
                self.printStack()
            return
        print('Stack is Empty!')

    def getSize(self):
        temp = Node()
        temp = self.top
        count = 0
        if not self.isStackEmpty():
            while temp:
                count += 1
                temp = temp.next
            return count
        print("Stack is Empty!")

    def printStack(self):
        temp = Node()
        temp = self.top
        if not self.isStackEmpty():
            while temp:
                print(temp.data, end=' ')
                temp = temp.next
            print('\n')
        else:
            print("Stack is Empty!")

if __name__ == '__main__':
    stack = Stack()
    stack.push(5, verbose=True)
    stack.push(6, verbose=True)
    stack.push(7, verbose=True)
    print("Size of the stack is %d" % stack.getSize())
    stack.pop(verbose=True)
    stack.pop(verbose=True)
    print("Size of the stack is %d" % stack.getSize())

File: DataStruct/c_cpp/stack.c
#include <stdio.h>
#include <string.h>
#define STACKSIZE 100

struct stack {
	int top;
	int items[STACKSIZE];	
};

int empty(struct stack *s)
{
	if(s->top == -1)
		return 1;
	return 0;	
}

int pop(struct stack *s){
	if(!empty(s)){
		return (s->items[s->top--]);
	}
	else
		printf("Underflow\n");
		return;	
}

int stackfull(struct stack *s){
	if(s->top == STACKSIZE - 1)
		return 1;
	return 0;	
}

void push(struct stack *s, int x){
	if(!stackfull(s)){
		s->items[++(s->top)] = x;
		return;
	}
	else{
		printf("stack overflow\n");
		return;
	}
}

void printStack(struct stack *s){
	int i;
	for (i=s->top; i>=0; i--){
		printf("%d ", s->items[i]);
	}
	printf("\n");
}

struct stack s;

int main(){
	s.top = -1;
	push(&s, 1);
	printStack(&s);
	push(&s, 5);
	printStack(&s);
	push(&s, 4);
	push(&s, 6);
	push(&s, 19);
	printStack(&s);
	push(&s, 18);
	printStack(&s);
	pop(&s);
	printStack(&s);
	return 0;
}